File: day:51/containDuplicate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let hash = {};
    
    for(let i = 0; i < nums.length; i++) {
        const num = (nums[i]).toString();
        if(hash[num] !== undefined) {
            if(Math.abs(hash[num] - i) <= k) {
                return true;
            }
            hash[num] = i;
        }
        else {
            hash[num] = i;
        }
    }
    return false;
    
};

File: day:51/containDuplicate.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hash = {};
    
        for i in range(len(nums)):
            num = str(nums[i]);
            if(num in hash):
                if(abs(hash[num] - i) <= k):
                    return True;
                hash[num] = i;
            else:
                hash[num] = i;
                
        return False;
